doc/source/whatsnew/v1.4.0.rst

```diff
@@ -187,7 +187,7 @@ Plotting
 
 Groupby/resample/rolling
 ^^^^^^^^^^^^^^^^^^^^^^^^
--
+- Bug in :meth:`GroupBy.shift` that would return different columns if ``fill_value`` was not None (:issue:`41556`)
```

Comment:
Can be a little more specific, different columns -> the group by columns

Reply:
Also - can you add a line in the performance section.

Reply:
Done